hazelcast-sql/src/test/java/com/hazelcast/sql/impl/client/SqlResubmissionStreamTest.java

```diff
@@ -0,0 +1,121 @@
+public class SqlResubmissionStreamTest extends SqlResubmissionTestSupport {
+    private static final int INITIAL_CLUSTER_SIZE = 1;
+    private static final int VALUES_PER_SECOND = 10000;
+
+    @Parameterized.Parameter
+    public ClusterFailureTestSupport.SingleFailingInstanceClusterFailure clusterFailure;
+
+    @Parameterized.Parameter(1)
+    public ClientSqlResubmissionMode resubmissionMode;
+
+    private HazelcastInstance client;
+
+    @Parameterized.Parameters(name = "clusterFailure:{0}, mode:{1}")
+    public static Collection<Object[]> parameters() {
+        List<Object[]> res = new ArrayList<>();
+        List<ClusterFailureTestSupport.SingleFailingInstanceClusterFailure> failures = Arrays.asList(
+                new ClusterFailureTestSupport.NodeReplacementClusterFailure(),
+                new ClusterFailureTestSupport.NodeShutdownClusterFailure(),
+                new ClusterFailureTestSupport.NetworkProblemClusterFailure(),
+                new ClusterFailureTestSupport.NodeTerminationClusterFailure()
+        );
+        for (ClientSqlResubmissionMode mode : ClientSqlResubmissionMode.values()) {
+            for (ClusterFailureTestSupport.SingleFailingInstanceClusterFailure failure : failures) {
+                res.add(new Object[]{failure, mode});
+            }
+        }
+        return res;
+    }
+
+    @Before
+    public void initFailure() {
+        clusterFailure.initialize(INITIAL_CLUSTER_SIZE, smallInstanceConfig());
+        ClientConfig clientConfig = new ClientConfig();
+        clientConfig.getSqlConfig().setResubmissionMode(resubmissionMode);
+        client = clusterFailure.createClient(clientConfig);
+    }
+
+    @Test
+    public void when_failingSelectAfterSomeDataIsFetched() {
+        SqlStatement statement = new SqlStatement("select * from table(generate_stream(" + VALUES_PER_SECOND + "))");
+        SqlResult rows = client.getSql().execute(statement);
+
+        try {
+            // In this test we expect the rows to be: 0, 1, 2, ... X, 0, 1, 2, ... N.
+            boolean resubmitted = false;
+            long expectedValue = 0;
+            int rowsSeen = 0;
+            for (SqlRow r : rows) {
+                long rowValue = r.getObject(0);
+                if (rowsSeen++ == VALUES_PER_SECOND / 2) {
+                    clusterFailure.fail();
+                }
+                if (expectedValue > 0 && rowValue == 0) {
+                    assertFalse("rows restarted from 0 for the 2nd time", resubmitted);
+                    resubmitted = true;
+                    expectedValue = 0;
+                }
+                assertEquals(expectedValue, rowValue);
+                expectedValue++;
+                if (resubmitted) {
+                    break;
+                }
+            }
+            assertTrue("resubmission didn't happen", resubmitted);
+        } catch (HazelcastSqlException e) {
+            if (!shouldFailAfterSomeDataIsFetched(resubmissionMode)) {
```

Comment:
This is a bit hidden, I was wondering why the test is supposed to pass with `ClientSqlResubmissionMode#NEVER` before I noticed this.

Maybe put a comment at the top?